### PR TITLE
Manually created sitemap.xml for improved control over indexed docs pages

### PR DESCRIPTION
**Pull request by DimedS**
## Description

Addresses #3741 by manually creating a `sitemap.xml` file instead of relying on the auto-generated version. This change is intended as a test to determine if it improves our desired outcomes. Depending on the results, we will proceed with the next steps as outlined in [this comment](https://github.com/kedro-org/kedro/issues/3741#issuecomment-2331062952).

## Developer Certificate of Origin
We need all contributions to comply with the [Developer Certificate of Origin (DCO)](https://developercertificate.org/). All commits must be signed off by including a `Signed-off-by` line in the commit message. [See our wiki for guidance](https://github.com/kedro-org/kedro/wiki/Guidelines-for-contributing-developers/).

If your PR is blocked due to unsigned commits, then you must follow the instructions under "Rebase the branch" on the GitHub Checks page for your PR. This will retroactively add the sign-off to all unsigned commits and allow the DCO check to pass.

## Checklist

- [ ] Read the [contributing](https://github.com/kedro-org/kedro/blob/main/CONTRIBUTING.md) guidelines
- [ ] Signed off each commit with a [Developer Certificate of Origin (DCO)](https://docs.github.com/en/repositories/managing-your-repositorys-settings-and-features/managing-repository-settings/managing-the-commit-signoff-policy-for-your-repository)
- [ ] Opened this PR as a 'Draft Pull Request' if it is work-in-progress
- [ ] Updated the documentation to reflect the code changes
- [ ] Added a description of this change in the [`RELEASE.md`](https://github.com/kedro-org/kedro/blob/main/RELEASE.md) file
- [ ] Added tests to cover my changes
- [ ] Checked if this change will affect Kedro-Viz, and if so, communicated that with the Viz team

### CODE REVIEW

File: docs/source/sitemap.xml [hunk 1]
```diff
@@ -0,0 +1,14 @@
...
+    <lastmod>2024-08-01T18:53:11.571849+00:00</lastmod>
+    <changefreq>monthly</changefreq>
+    <priority>1.0</priority>
+  </url>
+  <url>
+    <loc>https://docs.kedro.org/en/latest/</loc>
+    <lastmod>2024-08-09T09:39:27.628501+00:00</lastmod>
+    <changefreq>daily</changefreq>
+    <priority>0.5</priority>
+  </url>
```
Comment (astrojuanlu) [inline docs/source/sitemap.xml:13]: Can we include the subprojects as well, given that they aren't included in the sitemap https://github.com/readthedocs/readthedocs.org/issues/6841? hence

- https://docs.kedro.org/projects/kedro-viz/en/stable/
- https://docs.kedro.org/projects/kedro-viz/en/latest/
- https://docs.kedro.org/projects/kedro-datasets/en/kedro-datasets-4.1.0/ (unfortunately there's no  `stable` tag for kedro-datasets, see https://github.com/readthedocs/readthedocs.org/issues/5319)
- https://docs.kedro.org/projects/kedro-datasets/en/latest/

**Review by astrojuanlu — COMMENTED** [review]
Thanks @DimedS! Left a few comments

File: docs/source/sitemap.xml (hunk 1 above)
Comment (astrojuanlu) [reply docs/source/sitemap.xml:13]: Actually I just implemented a `/stable` redirection for kedro-datasets

image

(more context https://github.com/readthedocs/readthedocs.org/issues/11183#issuecomment-2333405041)

So we can add

- https://docs.kedro.org/projects/kedro-datasets/en/stable/

Comment (DimedS) [reply docs/source/sitemap.xml:34]: Many thanks @astrojuanlu , the redirection works well. I agree, projects added

File: docs/source/sitemap.xml [hunk 2]
```diff
@@ -0,0 +1,14 @@
+<urlset xmlns="http://www.sitemaps.org/schemas/sitemap/0.9" xmlns:xhtml="http://www.w3.org/1999/xhtml">
+  <url>
+    <loc>https://docs.kedro.org/en/stable/</loc>
+    <lastmod>2024-08-01T18:53:11.571849+00:00</lastmod>
+    <changefreq>monthly</changefreq>
+    <priority>1.0</priority>
+  </url>
+  <url>
+    <loc>https://docs.kedro.org/en/latest/</loc>
+    <lastmod>2024-08-09T09:39:27.628501+00:00</lastmod>
```
[the remark this answers is not in the document]
Comment (DimedS) [reply docs/source/sitemap.xml]: Makes sense, removed them from all latest version for kedro and projects

**Review by astrojuanlu — APPROVED** [review]
🤞🏼 and awaiting for the next release...

**Comment by astrojuanlu** [comment]
Thanks a lot @DmitrySorokinQB!

**Review by merelcht — APPROVED** [review]
I don't know much about sitemaps, but this looks sensible to me 👍

File: docs/source/sitemap.xml [hunk 3]
```diff
@@ -0,0 +1,35 @@
+<urlset xmlns="http://www.sitemaps.org/schemas/sitemap/0.9" xmlns:xhtml="http://www.w3.org/1999/xhtml">
+  <url>
+    <loc>https://docs.kedro.org/en/stable/</loc>
+    <lastmod>2024-09-01T18:53:11.571849+00:00</lastmod>
```
Comment (astrojuanlu) [inline docs/source/sitemap.xml:4]: These dates have to be put here by hand, but we can try to guesstimate them so they are a bit more realistic @merelcht

Comment (merelcht) [reply docs/source/sitemap.xml:4]: @lrcouto might be best if you update this and merge right before the release.

**Comment by merelcht** [comment]
Wait was the date updated in the end or not? @lrcouto @ElenaKhaustova @astrojuanlu

**Comment by astrojuanlu** [comment]
More importantly, the sitemap hasn't changed... https://github.com/kedro-org/kedro/issues/3741#issuecomment-2406018684

**Pull request merged**